faunadb-scala/src/main/scala/faunadb/values/Codec.scala

```diff
@@ -145,11 +196,20 @@ object Decoder {
       }
   }
 
-  implicit object ByteArrayDecoder extends Decoder[Array[Byte]] {
+  implicit def OptionDecoder[T](implicit decoder: Decoder[T]): Decoder[Option[T]] = new Decoder[Option[T]] {
     def decode(v: Value, path: FieldPath) =
       v match {
-        case BytesV(b) => Result.successful(b, path)
-        case v => Result.Unexpected(v, "Bytes", path)
+        case NullV => Result.successful(None, path)
+        case _ => decoder.decode(v, path).map { Some(_) }
```

Comment:
For single line maps, unless you need a `case`, prefer parentheses. Like `decoder.decode(v, path) map (Some(_))`